nebari-dev/nebari
# Disable `jupyterlab-jhub-apps` extension when jhub-apps is disabled

## Reference Issues or PRs

- Sibling to https://github.com/nebari-dev/nebari-docker-images/pull/185
- Addresses https://github.com/nebari-dev/jhub-apps/issues/472
- https://github.com/nebari-dev/jupyterlab-jhub-apps

Reference for the `labconfig/page_config.json` file:
- https://jupyterlab.readthedocs.io/en/stable/user/extensions.html#enable-disable-config
- https://jupyterlab.readthedocs.io/en/stable/user/directories.html#labconfig-directories

## What does this implement/fix?

_Put a `x` in the boxes that apply_


## Testing

     Yes

## How to test this PR?

1. Deploy with:
  ```yaml
  default_images:
    jupyterhub: quay.io/nebari/nebari-jupyterhub:jhub-apps-2024-10-1-and-jlab-ext
    jupyterlab: quay.io/nebari/nebari-jupyterlab:jhub-apps-2024-10-1-and-jlab-ext
  jhub_apps:
    enabled: true
  ```
2. Open JupyterLab
3. Open a Notebook
4. See that there is a toolbar item ":rocket: Deploy App" like this:
   ![image](https://github.com/user-attachments/assets/9c2baf57-bffc-4e33-874e-9d5e9ef49ccb)
5. Switch `jhub_apps.enabled` to `false` and deploy again
6. Open JupyterLab/Notebook see that the toolbar item is no longer here

## Any other comments?

None
